server/server.go

```diff
@@ -108,9 +112,18 @@ func (s *Server) handleConnection(conn *net.TCPConn) {
 			s.sendMessage(reply)
 		}
 	}
+	// If the protocol just became in use, save the session
+	if !activeBefore && sess.Proto.IsActive() {
+		if !sess.Save() {
+			log.Panicln("Session was not saved to the database")
+		}
+	}
+
 	if sess.Proto.IsActive() && dec[0] != nil {
 		// Print the decoded message and IP
 		fmt.Printf("%s: %s\n", msg.SourceIP, dec[0])
+		// TODO save message to DB
+		db.InsertMessage(sess.Proto.GetSessionID(), string(dec[0]), time.Now().Format(time.RFC3339), 1)
```

Comment:
In #7 , we will move this behavior to `ChatMessage`, again, let's leave it for now as it's a reminder of what used to be done